flujo/application/core/executor_core.py

```
@@ -743,6 +754,97 @@ async def _execute_complex_step(
+            return
+
+        # Create a copy to avoid modification during iteration if tasks complete
+        pending = list(self._background_tasks)
+        if not pending:
+            return
+
+        telemetry.logfire.info(
+            f"Waiting for {len(pending)} background tasks to complete (timeout={timeout}s)..."
+        )
+        try:
+            done, pending_set = await asyncio.wait(
```

Ruff (0.14.5)

Unpacked variable `done` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)